### PR TITLE
added proper update, renamed old update to saveExisting. wrote very m…

### PR DESCRIPTION
**Pull request by niels-qup**
…inimal test (will add integration tests later)

### CODE REVIEW

File: database/dynamo/transaction_writer.go [hunk 1]
```diff
@@ -77,7 +77,37 @@ func (b TransactionWriter) Create(record interface{}) TransactionWriter {
...
+
+	mapped, err := b.Encoder.MarshalMap(values)
+
+	if err != nil {
+		return b.addError(err)
+	}
+
+	query := dynamodb.TransactWriteItem{
+		Update: &dynamodb.Update{
+			ExpressionAttributeValues: mapped,
```
Comment (pcittadini) [inline database/dynamo/transaction_writer.go:96]: the only problem I can see here is the case when an attribute name is using some reserved dynamo name (ie. status) in that case we would need to inject something like:

```
ExpressionAttributeNames: map[string]*string{
			"#st": aws.String("status"),
		},
```

to create an alias.

File: reflection/tag.go [hunk 2]
```diff
@@ -7,7 +7,7 @@ import (
 
 func validStruct(typeReflection reflect.Type) (reflect.Type, error) {
 	if typeReflection.Kind() == reflect.Ptr {
-		typeReflection = typeReflection.Elem()
+		return validStruct(typeReflection.Elem())
```
Comment (niels-qup) [inline reflection/tag.go:10]: Useful for pointer of pointer of structs or deeper

**Review by pcittadini — APPROVED** [review]
so far lgtm

File: database/dynamo/transaction_writer.go (hunk 1 above)
Comment (niels-qup) [reply database/dynamo/transaction_writer.go:96]: Yeah that case has not been covered by the entire library yet, it's on the roadmap to program in ExpressionAttributeNames and Values separately

Comment (niels-qup) [reply database/dynamo/transaction_writer.go:96]: Created task https://mrticket.atlassian.net/browse/QT-413 to address this problem over the entire lib.
Merging this one.

**Pull request merged**